rhel/vex/fetcher.go

```diff
@@ -0,0 +1,288 @@
+		uri, err := u.url.Parse(compressedFilename)
+		if err != nil {
+			return nil, hint, err
+		}
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, uri.String(), nil)
+		if err != nil {
+			return nil, hint, err
+		}
+
+		u.client.Timeout = time.Minute * 2 // This is a large request.
+		res, err := u.client.Do(req)
+		if res != nil {
+			defer res.Body.Close()
+		}
+		if err != nil {
+			return nil, hint, err
+		}
+		if res.StatusCode != http.StatusOK {
+			return nil, hint, fmt.Errorf("unexpected response from latest compressed file: %s", res.Status)
+		}
+
+		lm := res.Header.Get("last-modified")
+		fp.requestTime, err = time.Parse(http.TimeFormat, lm)
+		if err != nil {
+			return nil, hint, fmt.Errorf("could not parse last-modified header %s:, %w", lm, err)
+		}
+		z, err := zstd.NewReader(res.Body)
+		if err != nil {
+			return nil, hint, err
+		}
+		defer z.Close()
+		r := tar.NewReader(z)
+
+		var (
+			h              *tar.Header
+			buf, bc        bytes.Buffer
+			entriesWritten int
+		)
+		for h, err = r.Next(); err == nil; h, err = r.Next() {
+			if h.Typeflag != tar.TypeReg {
+				continue
+			}
+			year, err := strconv.ParseInt(path.Dir(h.Name), 10, 64)
+			if err != nil {
+				return nil, hint, fmt.Errorf("error parsing year %w", err)
+			}
+			if year < lookBackToYear {
+				continue
+			}
+			buf.Grow(int(h.Size))
+			if _, err := buf.ReadFrom(r); err != nil {
+				return nil, hint, err
+			}
+
+			err = json.Compact(&bc, buf.Bytes())
+			if err != nil {
+				return nil, hint, fmt.Errorf("error compressing JSON %s: %w", h.Name, err)
+			}
+			bc.WriteByte('\n')
+			cw.Write(bc.Bytes())
+			buf.Reset()
+			bc.Reset()
+			entriesWritten++
+		}
+		if err != io.EOF {
+			return nil, hint, fmt.Errorf("error reading tar contents: %w", err)
+		}
+
+		zlog.Debug(ctx).
+			Str("updater", u.Name()).
+			Int("entries written", entriesWritten).
+			Msg("finished writing compressed data to spool")
+
+	}
+
+	uri, err := u.url.Parse("changes.csv")
+	if err != nil {
+		return nil, hint, err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, uri.String(), nil)
+	if err != nil {
+		return nil, hint, err
+	}
+	if fp.changesEtag != "" {
+		req.Header.Add("If-None-Match", fp.changesEtag)
+	}
+	res, err := u.client.Do(req)
+	if res != nil {
+		defer res.Body.Close()
+	}
+	if err != nil {
+		return nil, hint, err
+	}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		if t := fp.changesEtag; t == "" || t != res.Header.Get("etag") {
+			break
+		}
+		fallthrough
+	case http.StatusNotModified:
+		// We could return driver.Unchanged here but we don't know for sure. Return the
+		// file that may have data read from the compressed file in it.
+		return f, hint, nil
+	default:
+		return nil, hint, fmt.Errorf("unexpected response from changes.csv: %s", res.Status)
+	}
+	fp.changesEtag = res.Header.Get("etag")
+
+	rd := csv.NewReader(res.Body)
+	rd.FieldsPerRecord = 2
+	rd.ReuseRecord = true
+	var (
+		l       = 0
```

Comment:
curious why not just `int`?

Reply:
for me it's marginally clearer this way, can change, what is the argument for the other way?